File: 7-kyu/initialize-my-name/python/solution.py
def initialize_names(name):
    name = name.split()
    if len(name) == 1:
        return name[0]
    elif len(name) == 2:
        return ' '.join(name)
    return f'{name[0]} {". ".join(n[0] for n in name[1:-1])}. {name[-1]}'

File: 7-kyu/initialize-my-name/javascript/solution.js
const initializeNames = name => {
  const nameSplit = name.split(" ");
  switch (nameSplit.length) {
      case 0:
      return "";
      case 1:
      case 2:
      return name;
      default:
      const middleInitials = nameSplit.slice(1, -1).map(n => `${n[0]}. `).join("");
      return `${nameSplit[0]} ${middleInitials}${nameSplit[nameSplit.length - 1]}`;
  }
};
